lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -37,237 +39,1042 @@
+        template_conf.update({'vrfSegmentId': create['vrf_id']})
+        template_conf.update({'vrfName': create['vrf_name']})
+
+        create_upd.update({'vrfTemplateConfig': json.dumps(template_conf)})
 
-    return False
+        return create_upd
 
 
-def vrf_create_payload(module):
+    def get_have(self):
+        """
 
-    payload = dict()
-    payload['fabric'] = module.params['fabric']
-    payload['vrfName'] = module.params['vrf_name']
-    payload['vrfTemplate'] = module.params['vrf_template']
-    payload['vrfExtensionTemplate'] = module.params['vrf_extension_template']
-    payload['vrfTemplateConfig'] = module.params['vrf_template_config']
-    payload['vrfId'] = module.params['vrf_id']
+        Sufficient error checking??
+        Possiibe inputs from DCNM:
+        1. API failures
+        2. Blank responses - no vrfs
+                           - no attachments
+                           - no deployments
+        3. typecast booleans before checks?
+        """
 
-    json_data = json.dumps(payload)
+        have_create = list()
+        have_attach = list()
+        have_deploy = dict()
 
-    return json_data
+        all_vrfs = ''
 
+        method = 'GET'
+        path = '/rest/top-down/fabrics/{}/vrfs'.format(self.fabric)
+
+        resp = dcnm_send(self.module, method, path)
+
+        if resp.get('ERROR') == 'Not Found' and resp.get('RETURN_CODE') == 404:
+            logit("Fabric not found")
+            return have_create, have_attach, have_deploy
+
+        if not resp['DATA']:
+            return have_create, have_attach, have_deploy
+
+        for vrf in resp['DATA']:
+            all_vrfs += vrf['vrfName'] + ','
+
+        path = '/rest/top-down/fabrics/{}/vrfs/attachments?vrf-names={}'.format(self.fabric, all_vrfs[:-1])
+
+        all_vrf_attach = dcnm_send(self.module, method, path)
+
+        for vrf in resp['DATA']:
+            t_conf = dict()
+            t_conf.update({'vrfSegmentId': vrf['vrfId']})
+            t_conf.update({'vrfName': vrf['vrfName']})
+
+            vrf.update({'vrfTemplateConfig': json.dumps(t_conf)})
+            del vrf['vrfStatus']
+            have_create.append(vrf)
+
+        all_vrfs = ''
+
+        for vrf_attach in all_vrf_attach['DATA']:
+            if not vrf_attach.get('lanAttachList'):
+                continue
+            attach_list = vrf_attach['lanAttachList']
+            dep_vrf = ''
+            for attach in attach_list:
+                attach_state = False if attach['lanAttachState'] == "NA" else True
+                deploy = attach['isLanAttached']
+                if bool(deploy) and (attach['lanAttachState'] == "OUT-OF-SYNC" or
+                                     attach['lanAttachState'] == "PENDING"):
+                    deploy = False
+
+                if bool(deploy):
+                    dep_vrf = attach['vrfName']
+
+                sn = attach['switchSerialNo']
+                vlan = attach['vlanId']
+                del attach['vlanId']
```

Comment:
Can you add some comments around why you are deleting data and then updating data in the `attach` dict?

Reply:
Yeah, will do, its just to keep the data in the outgoing payload format. ex: vlanId in the incoming payload needs to be changed to just vlan in the outgoing payload.